### ISSUE TITLE
Would it make sense to allow fine-tuning specific layers only?

### ISSUE THREAD

**Issue by iivalchev** [LennartPurucker/finetune_tabpfn_v2#7]
Does it make sense to allow all model layers to be trainable during fine-tuning? And how would these relate to the size of the fine-tuning dataset? Perhaps we can enable the user to specify which parts of the model to be trained, like encoders-only, attention-only, decoder-only? Is there a way to evaluate what should be tried first?

**Comment by LennartPurucker** [comment]
I played around with this a bit in the past. My intuition without hard data as a result:
* Only the transformer layers get most of the signal.
* The encode/decoder layers have a minor impact, and if they have an impact, they might just overfit. 
* For such a small model, this would only be really important if it improves the gradient flow, but I do not think that the gradient flow is improved by freezing certain layers.

**Comment by iivalchev** [comment]
I give it a shot when ready with my dataset I would. And will report my findings. Many thanks @LennartPurucker
